packages/web3-utils/src/converters.ts

```diff
@@ -0,0 +1,340 @@
+	grand: expo10(21),
+	mether: expo10(24),
+	gether: expo10(27),
+	tether: expo10(30),
+};
+
+const errorMessage = (value: unknown, error: string) =>
+	`Invalid value given "${String(value)}". Error: ${error}.`;
+
+export type EtherUnits = keyof typeof ethUnitMap;
+
+/** @internal */
+const bytesToBuffer = (data: Bytes): Buffer | never => {
+	if (Buffer.isBuffer(data)) {
+		return data;
+	}
+
+	if (data instanceof Uint8Array) {
+		return Buffer.from(data);
+	}
+
+	if (Array.isArray(data)) {
+		if (data.some(d => d < 0)) {
+			throw new Error(errorMessage(data, 'contains negative values'));
+		}
+
+		if (data.some(d => d > 255)) {
+			throw new Error(errorMessage(data, 'contains numbers greater than 255'));
+		}
+
+		if (data.some(d => !Number.isInteger(d))) {
+			throw new Error(errorMessage(data, 'contains invalid integer values'));
+		}
+
+		return Buffer.from(data);
+	}
+
+	if (typeof data === 'string' && !isHexStrict(data)) {
+		throw new Error(errorMessage(data, 'not valid hex string'));
+	}
+
+	if (typeof data === 'string' && (data.startsWith('0x') || data.startsWith('0X'))) {
+		return Buffer.from(data.substr(2), 'hex');
+	}
+
+	throw new Error(errorMessage(data, 'can not parse as byte data'));
+};
+
+/** @internal */
+const bufferToHexString = (data: Buffer) => `0x${data.toString('hex')}`;
+
+/**
+ * Convert a byte array to a hex string
+ */
+export const bytesToHex = (bytes: Bytes): HexString => bufferToHexString(bytesToBuffer(bytes));
+
+/**
+ * Convert a hex string to a byte array
+ */
+export const hexToBytes = (bytes: HexString): Buffer => bytesToBuffer(bytes);
+
+/**
+ * Converts value to it's number representation
+ */
+export const hexToNumber = (value: HexString): bigint | number => {
+	if (!isHexStrict(value)) {
+		throw new Error(errorMessage(value, 'not valid hex string'));
+	}
+
+	const [negative, hexValue] = value.startsWith('-') ? [true, value.substr(1)] : [false, value];
+	const num = BigInt(hexValue);
+
+	if (num > BigInt(Number.MAX_SAFE_INTEGER) || num < BigInt(Number.MIN_SAFE_INTEGER)) {
+		return negative ? BigInt(-1) * BigInt(num) : BigInt(num);
+	}
+
+	return negative ? -1 * Number(num) : Number(num);
+};
+
+/**
+ * Converts value to it's number representation @alias `hexToNumber`
+ */
+export const toDecimal = hexToNumber;
+
+/**
+ * Converts value to it's hex representation
+ */
+export const numberToHex = (value: Numbers): HexString => {
+	try {
+		if (typeof value === 'number' && !Number.isFinite(value)) {
+			throw new Error(errorMessage(value, 'contains invalid integer values'));
+		}
+
+		if ((typeof value === 'number' || typeof value === 'bigint') && BigInt(value) < BigInt(0)) {
+			return `-0x${value.toString(16).substr(1)}`;
+		}
+
+		if (
+			(typeof value === 'number' || typeof value === 'bigint') &&
+			BigInt(value) >= BigInt(0)
+		) {
+			return `0x${value.toString(16)}`;
+		}
+
+		if (typeof value === 'string' && isHexStrict(value)) {
+			return numberToHex(hexToNumber(value));
+		}
```

Comment:
Can't we just return here? If `value` passes `isHexStrict` than it's already a prefixed hex string

Reply:
Yes we can this check make sure the sanity of the input. It its actually a number.